omp/distributed/index_map_kernels.cpp

```diff
@@ -239,6 +239,77 @@ GKO_INSTANTIATE_FOR_EACH_LOCAL_GLOBAL_INDEX_TYPE(
     GKO_DECLARE_INDEX_MAP_MAP_TO_LOCAL);
 
 
+template <typename LocalIndexType, typename GlobalIndexType>
+void map_to_global(
+    std::shared_ptr<const DefaultExecutor> exec,
+    device_partition<const LocalIndexType, const GlobalIndexType> partition,
+    device_segmented_array<const GlobalIndexType> remote_global_idxs,
+    experimental::distributed::comm_index_type rank,
+    const array<LocalIndexType>& local_ids,
+    experimental::distributed::index_space is,
+    array<GlobalIndexType>& global_ids)
+{
+    const auto& ranges_by_part = partition.ranges_by_part;
+    auto local_ranges = ranges_by_part.get_segment(rank);
+
+    global_ids.resize_and_reset(local_ids.get_size());
+
+    auto local_size =
+        static_cast<LocalIndexType>(partition.part_sizes_begin[rank]);
+    auto remote_size = static_cast<LocalIndexType>(
+        remote_global_idxs.flat_end - remote_global_idxs.flat_begin);
+    size_type local_range_id = 0;
+    if (is == experimental::distributed::index_space::local) {
```

Comment:
same here, I think this might be more readable pulling the condition in? Also can we copy the approach from `reference`, since it uses named lambdas instead of unnamed pieces of code

Reply:
I think one compiler didn't work with lambdas within the parallel for loop, maybe nvhpc. I will still hoist the ifs into the loop.

Reply:
Intel's classic compiler used to have the opposite issue (you needed to put the loop body into an IIFE to work around some compiler crashes). Then maybe the code could use some comments describing the different cases briefly